File: backend/Models/SavedCandidate.cs
namespace LocalHire.Api.Models;

public sealed class SavedCandidate
{
    public Guid Id { get; set; }
    public Guid EmployerId { get; set; }
    public UserRole EmployerRole { get; set; } = UserRole.Hiring;
    public Guid WorkerId { get; set; }
    public UserRole WorkerRole { get; set; } = UserRole.LookingForWork;
    public DateTimeOffset CreatedAt { get; set; }

    public User Employer { get; set; } = null!;
    public User Worker { get; set; } = null!;
}

File: integration-tests/features/db_models/saved_candidate.py
"""Row model for the ``SavedCandidates`` table."""

from dataclasses import dataclass, field
from datetime import datetime

from features.db_models.model import Model
from features.utils.util import utc_now


@dataclass
class SavedCandidate(Model):
    id: str
    employer_id: str
    worker_id: str
    employer_role: str = "Hiring"
    worker_role: str = "LookingForWork"
    created_at: datetime = field(default_factory=utc_now)
